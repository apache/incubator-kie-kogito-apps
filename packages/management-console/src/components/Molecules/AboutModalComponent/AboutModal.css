.pf-c-about-modal-box__hero {
  background-attachment: unset;
}
