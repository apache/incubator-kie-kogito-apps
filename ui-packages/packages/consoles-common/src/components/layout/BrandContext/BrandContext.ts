import React, { useContext } from 'react';

export interface Brand {
  imageSrc: string;
  altText: string;
}

export const BrandContext = React.createContext<Brand>(null);

export const useBrandContext = (): Brand => useContext<Brand>(BrandContext);
