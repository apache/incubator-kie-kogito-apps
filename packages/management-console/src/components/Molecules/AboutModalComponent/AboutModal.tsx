import React, { useState } from 'react';
import { AboutModal, Button, TextContent, TextList, TextListItem } from '@patternfly/react-core';
import './AboutModal.css';
export interface IOwnProps {
  isOpenProp: boolean;
  handleModalToggleProp: any;
}

const AboutModalBox: React.FC<IOwnProps> = ({ isOpenProp, handleModalToggleProp }) => {
  return (
    <AboutModal
      isOpen={isOpenProp}
      onClose={handleModalToggleProp}
      trademark="Trademark and copyright information here"
      brandImageAlt="Kogito Logo"
      brandImageSrc={require('../../../static/kogito_about_logo.png')}
      productName="Kogito"
    >
      <TextContent>
        <TextList component="dl" style={{ width: '100%' }}>
          <TextListItem component="dt">CFME Version</TextListItem>
          <TextListItem component="dd">5.5.3.4.20102789036450</TextListItem>
          <TextListItem component="dt">Cloudforms Version</TextListItem>
          <TextListItem component="dd">4.1</TextListItem>
          <TextListItem component="dt">Server Name</TextListItem>
          <TextListItem component="dd">40DemoMaster</TextListItem>
          <TextListItem component="dt">User Name</TextListItem>
          <TextListItem component="dd">Administrator</TextListItem>
          <TextListItem component="dt">User Role</TextListItem>
          <TextListItem component="dd">EvmRole-super_administrator</TextListItem>
          <TextListItem component="dt">Browser Version</TextListItem>
          <TextListItem component="dd">601.2</TextListItem>
          <TextListItem component="dt">Browser OS</TextListItem>
          <TextListItem component="dd">Mac</TextListItem>
        </TextList>
      </TextContent>
    </AboutModal>
  );
};

export default AboutModalBox;
