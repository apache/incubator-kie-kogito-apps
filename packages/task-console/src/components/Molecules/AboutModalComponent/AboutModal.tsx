import React from 'react';
import {
  AboutModal,
  TextContent,
  Text,
  TextList,
  TextListItem
} from '@patternfly/react-core';
import './AboutModal.css';
// import aboutPageLogo from '../../../static/managementConsoleLogo.svg';
import aboutPageLogo from '../../../static/kogitoLogo.png';
import aboutPageBackground from '../../../static/KogitoAbout.png';
import { version } from '../../../../package.json';
export interface IOwnProps {
  isOpenProp: boolean;
  handleModalToggleProp: any;
}
const AboutModalBox: React.FC<IOwnProps> = ({
  isOpenProp,
  handleModalToggleProp
}) => {
  return (
    <AboutModal
      isOpen={isOpenProp}
      onClose={handleModalToggleProp}
      trademark="Task Console is part of Kogito, an open source software released under the Apache Software License 2.0"
      brandImageAlt="Kogito Logo"
      brandImageSrc={aboutPageLogo}
      backgroundImageSrc={aboutPageBackground}
    >
      <TextContent>
        <Text component="h5" />
        <TextList component="dl">
          <TextListItem component="dt">Version: </TextListItem>
          <TextListItem component="dd">{version}</TextListItem>
          <TextListItem component="dt">License information: </TextListItem>
          <TextListItem component="dd">
            <a
              href="https://github.com/kiegroup/kogito-runtimes/blob/master/LICENSE"
              target="_blank"
            >
              https://github.com/kiegroup/kogito-runtimes/blob/master/LICENSE
            </a>
          </TextListItem>
          <TextListItem component="dt">Report a bug: </TextListItem>
          <TextListItem component="dd">
            <a href="https://issues.redhat.com/projects/KOGITO" target="_blank">
              https://issues.redhat.com/projects/KOGITO
            </a>
          </TextListItem>
          <TextListItem component="dt">Get involved/help/docs: </TextListItem>
          <TextListItem component="dd">
            <a
              href="https://github.com/kiegroup/kogito-runtimes/wiki"
              target="_blank"
            >
              https://github.com/kiegroup/kogito-runtimes/wiki
            </a>
          </TextListItem>
          <TextListItem component="dt">Kogito URL: </TextListItem>
          <TextListItem component="dd">
            <a href="http://kogito.kie.org" target="_blank">
              http://kogito.kie.org
            </a>
          </TextListItem>
          <TextListItem component="dt">Data-Index URL: </TextListItem>
          <TextListItem component="dd">
            <a
              href={process.env.KOGITO_DATAINDEX_URL + '/graphql'}
              target="_blank"
            >
              {process.env.KOGITO_DATAINDEX_HTTP_URL + '/graphql'}
            </a>
          </TextListItem>
        </TextList>
      </TextContent>
    </AboutModal>
  );
};

export default AboutModalBox;
