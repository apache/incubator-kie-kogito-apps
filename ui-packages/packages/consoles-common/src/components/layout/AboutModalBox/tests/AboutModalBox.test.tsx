import React from 'react';
import { render, screen } from '@testing-library/react';
import AboutModal from '../AboutModalBox';
import { BrandContext } from '../../BrandContext/BrandContext';

const props = {
  isOpenProp: true,
  handleModalToggleProp: jest.fn()
};
describe('AboutModal component tests', () => {
  it('snapshot testing', () => {
    process.env.KOGITO_APP_VERSION = '1.2.3-MOCKED-VERSION';
    const { container } = render(
      <BrandContext.Provider
        value={{
          imageSrc: 'kogito-image-src',
          altText: 'kogito image alt text'
        }}
      >
        <AboutModal {...props} />
      </BrandContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
