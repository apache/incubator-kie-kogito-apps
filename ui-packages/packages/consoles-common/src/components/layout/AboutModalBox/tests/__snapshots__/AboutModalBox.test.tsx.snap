// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutModal component tests snapshot testing 1`] = `
<AboutModal
  appendTo={null}
  backgroundImageSrc="kogitoAbout.png"
  brandImageAlt="kogito image alt text"
  brandImageSrc="kogito-image-src"
  className="kogito-consoles-common--aboutModalBox"
  data-ouia-component-type="AboutModalBox"
  data-ouia-safe={true}
  isOpen={true}
  noAboutModalBoxContentContainer={false}
  onClose={[MockFunction]}
  productName=""
  trademark="undefined is part of Kogito, an open source software released under the Apache Software License 2.0"
>
  <Portal
    containerInfo={
      <div>
        <div
          class="pf-c-backdrop"
        >
          <div
            class="pf-l-bullseye"
          >
            <div
              aria-describedby="pf-about-modal-content-0"
              aria-labelledby="pf-about-modal-title-0"
              aria-modal="true"
              class="pf-c-about-modal-box kogito-consoles-common--aboutModalBox"
              role="dialog"
            >
              <div
                class="pf-c-about-modal-box__brand"
              >
                <img
                  alt="kogito image alt text"
                  class="pf-c-about-modal-box__brand-image"
                  src="kogito-image-src"
                />
              </div>
              <div
                class="pf-c-about-modal-box__close"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close Dialog"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="OUIA-Generated-Button-plain-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </div>
              
              <div
                class="pf-c-about-modal-box__content"
                data-ouia-component-type="AboutModalBox"
                data-ouia-safe="true"
                id="pf-about-modal-content-0"
              >
                <div
                  class="pf-c-about-modal-box__body"
                >
                  <div
                    class="pf-c-content"
                  >
                    <div
                      class="pf-c-content"
                    >
                      <h5
                        class=""
                        data-pf-content="true"
                      />
                      <dl
                        class=""
                        data-pf-content="true"
                      >
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Version: 
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          1.2.3-MOCKED-VERSION
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          License information: 
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          <a
                            href="https://github.com/kiegroup/kogito-runtimes/blob/main/LICENSE"
                            rel="noreferrer"
                            target="_blank"
                          >
                            https://github.com/kiegroup/kogito-runtimes/blob/main/LICENSE
                          </a>
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Report a bug: 
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          <a
                            href="https://issues.redhat.com/projects/KOGITO"
                            rel="noreferrer"
                            target="_blank"
                          >
                            https://issues.redhat.com/projects/KOGITO
                          </a>
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Get involved/help/docs: 
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          <a
                            href="https://docs.jboss.org/kogito/release/latest/html_single/"
                            rel="noreferrer"
                            target="_blank"
                          >
                            https://docs.jboss.org/kogito/release/latest/html_single/
                          </a>
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Kogito URL: 
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          <a
                            href="https://kogito.kie.org"
                            rel="noreferrer"
                            target="_blank"
                          >
                            https://kogito.kie.org
                          </a>
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Data-Index URL: 
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                           
                        </dd>
                      </dl>
                    </div>
                  </div>
                </div>
                <p
                  class="pf-c-about-modal-box__strapline"
                >
                  undefined is part of Kogito, an open source software released under the Apache Software License 2.0
                </p>
              </div>
              <div
                class="pf-c-about-modal-box__hero"
                style="--pf-c-about-modal-box__hero--sm--BackgroundImage: url(kogitoAbout.png);"
              />
            </div>
          </div>
        </div>
      </div>
    }
  >
    <AboutModalContainer
      aboutModalBoxContentId="pf-about-modal-content-0"
      aboutModalBoxHeaderId="pf-about-modal-title-0"
      backgroundImageSrc="kogitoAbout.png"
      brandImageAlt="kogito image alt text"
      brandImageSrc="kogito-image-src"
      className="kogito-consoles-common--aboutModalBox"
      data-ouia-component-type="AboutModalBox"
      data-ouia-safe={true}
      isOpen={true}
      noAboutModalBoxContentContainer={false}
      onClose={[MockFunction]}
      productName=""
      trademark="undefined is part of Kogito, an open source software released under the Apache Software License 2.0"
    >
      <Backdrop>
        <div
          className="pf-c-backdrop"
        >
          <FocusTrap
            active={true}
            className="pf-l-bullseye"
            focusTrapOptions={
              Object {
                "clickOutsideDeactivates": true,
              }
            }
            paused={false}
          >
            <div
              className="pf-l-bullseye"
            >
              <AboutModalBox
                aria-describedby="pf-about-modal-content-0"
                aria-labelledby="pf-about-modal-title-0"
                className="kogito-consoles-common--aboutModalBox"
              >
                <div
                  aria-describedby="pf-about-modal-content-0"
                  aria-labelledby="pf-about-modal-title-0"
                  aria-modal="true"
                  className="pf-c-about-modal-box kogito-consoles-common--aboutModalBox"
                  role="dialog"
                >
                  <AboutModalBoxBrand
                    alt="kogito image alt text"
                    src="kogito-image-src"
                  >
                    <div
                      className="pf-c-about-modal-box__brand"
                    >
                      <img
                        alt="kogito image alt text"
                        className="pf-c-about-modal-box__brand-image"
                        src="kogito-image-src"
                      />
                    </div>
                  </AboutModalBoxBrand>
                  <AboutModalBoxCloseButton
                    onClose={[MockFunction]}
                  >
                    <div
                      className="pf-c-about-modal-box__close"
                    >
                      <Button
                        aria-label="Close Dialog"
                        onClick={[MockFunction]}
                        variant="plain"
                      >
                        <button
                          aria-disabled={false}
                          aria-label="Close Dialog"
                          className="pf-c-button pf-m-plain"
                          data-ouia-component-id="OUIA-Generated-Button-plain-1"
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe={true}
                          disabled={false}
                          onClick={[MockFunction]}
                          role={null}
                          type="button"
                        >
                          <TimesIcon
                            color="currentColor"
                            noVerticalAlign={false}
                            size="sm"
                          >
                            <svg
                              aria-hidden={true}
                              aria-labelledby={null}
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style={
                                Object {
                                  "verticalAlign": "-0.125em",
                                }
                              }
                              viewBox="0 0 352 512"
                              width="1em"
                            >
                              <path
                                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                              />
                            </svg>
                          </TimesIcon>
                        </button>
                      </Button>
                    </div>
                  </AboutModalBoxCloseButton>
                  <AboutModalBoxContent
                    data-ouia-component-type="AboutModalBox"
                    data-ouia-safe={true}
                    id="pf-about-modal-content-0"
                    noAboutModalBoxContentContainer={false}
                    trademark="undefined is part of Kogito, an open source software released under the Apache Software License 2.0"
                  >
                    <div
                      className="pf-c-about-modal-box__content"
                      data-ouia-component-type="AboutModalBox"
                      data-ouia-safe={true}
                      id="pf-about-modal-content-0"
                    >
                      <div
                        className="pf-c-about-modal-box__body"
                      >
                        <div
                          className="pf-c-content"
                        >
                          <TextContent>
                            <div
                              className="pf-c-content"
                            >
                              <Text
                                component="h5"
                              >
                                <h5
                                  className=""
                                  data-pf-content={true}
                                />
                              </Text>
                              <TextList
                                component="dl"
                              >
                                <dl
                                  className=""
                                  data-pf-content={true}
                                >
                                  <TextListItem
                                    component="dt"
                                  >
                                    <dt
                                      className=""
                                      data-pf-content={true}
                                    >
                                      Version: 
                                    </dt>
                                  </TextListItem>
                                  <TextListItem
                                    component="dd"
                                  >
                                    <dd
                                      className=""
                                      data-pf-content={true}
                                    >
                                      1.2.3-MOCKED-VERSION
                                    </dd>
                                  </TextListItem>
                                  <TextListItem
                                    component="dt"
                                  >
                                    <dt
                                      className=""
                                      data-pf-content={true}
                                    >
                                      License information: 
                                    </dt>
                                  </TextListItem>
                                  <TextListItem
                                    component="dd"
                                  >
                                    <dd
                                      className=""
                                      data-pf-content={true}
                                    >
                                      <a
                                        href="https://github.com/kiegroup/kogito-runtimes/blob/main/LICENSE"
                                        rel="noreferrer"
                                        target="_blank"
                                      >
                                        https://github.com/kiegroup/kogito-runtimes/blob/main/LICENSE
                                      </a>
                                    </dd>
                                  </TextListItem>
                                  <TextListItem
                                    component="dt"
                                  >
                                    <dt
                                      className=""
                                      data-pf-content={true}
                                    >
                                      Report a bug: 
                                    </dt>
                                  </TextListItem>
                                  <TextListItem
                                    component="dd"
                                  >
                                    <dd
                                      className=""
                                      data-pf-content={true}
                                    >
                                      <a
                                        href="https://issues.redhat.com/projects/KOGITO"
                                        rel="noreferrer"
                                        target="_blank"
                                      >
                                        https://issues.redhat.com/projects/KOGITO
                                      </a>
                                    </dd>
                                  </TextListItem>
                                  <TextListItem
                                    component="dt"
                                  >
                                    <dt
                                      className=""
                                      data-pf-content={true}
                                    >
                                      Get involved/help/docs: 
                                    </dt>
                                  </TextListItem>
                                  <TextListItem
                                    component="dd"
                                  >
                                    <dd
                                      className=""
                                      data-pf-content={true}
                                    >
                                      <a
                                        href="https://docs.jboss.org/kogito/release/latest/html_single/"
                                        rel="noreferrer"
                                        target="_blank"
                                      >
                                        https://docs.jboss.org/kogito/release/latest/html_single/
                                      </a>
                                    </dd>
                                  </TextListItem>
                                  <TextListItem
                                    component="dt"
                                  >
                                    <dt
                                      className=""
                                      data-pf-content={true}
                                    >
                                      Kogito URL: 
                                    </dt>
                                  </TextListItem>
                                  <TextListItem
                                    component="dd"
                                  >
                                    <dd
                                      className=""
                                      data-pf-content={true}
                                    >
                                      <a
                                        href="https://kogito.kie.org"
                                        rel="noreferrer"
                                        target="_blank"
                                      >
                                        https://kogito.kie.org
                                      </a>
                                    </dd>
                                  </TextListItem>
                                  <TextListItem
                                    component="dt"
                                  >
                                    <dt
                                      className=""
                                      data-pf-content={true}
                                    >
                                      Data-Index URL: 
                                    </dt>
                                  </TextListItem>
                                  <TextListItem
                                    component="dd"
                                  >
                                    <dd
                                      className=""
                                      data-pf-content={true}
                                    >
                                       
                                    </dd>
                                  </TextListItem>
                                </dl>
                              </TextList>
                            </div>
                          </TextContent>
                        </div>
                      </div>
                      <p
                        className="pf-c-about-modal-box__strapline"
                      >
                        undefined is part of Kogito, an open source software released under the Apache Software License 2.0
                      </p>
                    </div>
                  </AboutModalBoxContent>
                  <AboutModalBoxHero
                    backgroundImageSrc="kogitoAbout.png"
                  >
                    <div
                      className="pf-c-about-modal-box__hero"
                      style={
                        Object {
                          "--pf-c-about-modal-box__hero--sm--BackgroundImage": "url(kogitoAbout.png)",
                        }
                      }
                    />
                  </AboutModalBoxHero>
                </div>
              </AboutModalBox>
            </div>
          </FocusTrap>
        </div>
      </Backdrop>
    </AboutModalContainer>
  </Portal>
</AboutModal>
`;
